llvm-bazel/llvm-project-overlay/llvm/BUILD

```
@@ -453,13 +453,23 @@ cc_library(
     name = "tblgen",
     srcs = glob([
         "utils/TableGen/*.cpp",
-        "utils/TableGen/*.h",
         "utils/TableGen/GlobalISel/*.cpp",
+
+        # We have to include these headers here as well as in the `hdrs` below
+        # to allow the `.cpp` files to use file-relative-inclusion to find
+        # them, even though consumers of this library use inclusion relative to
+        # `utils/TableGen` with the `strip_includes_prefix` of this library.
+        # This mixture is likely incompatible with header modules.
```

"likely" incompatible? Seems like you turned it off so you concluded it was incompatible?

```suggestion
        # This mixture is incompatible with header modules.
```

Or are you not confident that there isn't some way to work around it and make it compatible?

```suggestion
        # This mixture appears to be incompatible with header modules.
```

---

(it was the latter)